src/pages/DrumMachinePage.js

```diff
@@ -93,31 +95,38 @@ export default function DrumMachinePage({ allPads }) {
     });
   }
   ////////////////////drumPad////////////////////
+
   ////////////////////record////////////////////
-  function recordStopClick() {
-    recorder.stop();
+  function recordStartClick() {
+    recorder.current.start();
   }
 
-  function recordStartClick() {
-    recorder.start();
+  function recordStopClick() {
+    recorder.current.stop();
+    loopPlayer.current.stop();
   }
   ////////////////////record////////////////////
+
   ////////////////////DrumLoop////////////////////
-  function startDrumLoop() {
-    if (isPlayin === playbutton) {
+  function startDrumLoop(isPlayin) {
+    if (isPlayin === false) {
       Tone.loaded().then(() => {
         loopPlayer.current.start();
       });
-      setIsPlayin(pausebutton);
     } else {
       loopPlayer.current.stop();
-      setIsPlayin(playbutton);
     }
   }
-  function getDrumLoop(event) {
-    setCurrentDrumLoop(event.target.value);
-    loopPlayer.current.stop();
-    setIsPlayin(playbutton);
+  function getDrumLoop(isPlayin, currentLoop) {
+    if (recorder.current.state === 'inactive') {
+      loopPlayer.current.stop()
+      setCurrentDrumLoop(currentLoop);
+    } else {
+      recordStopClick();
+      setCurrentDrumLoop(currentLoop);
+  
```

Comment:
Please check also here - setCurrentDrumLoop(currentLoop) is in both conditions - maybe you need to do it like this for coordination of drumloopplayer and record function, just to be sure :-)